main.go

```diff
@@ -4,30 +4,48 @@ import (
 	"github.com/terraform-linters/tflint-plugin-sdk/plugin"
 	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
 	"github.com/terraform-linters/tflint-ruleset-terraform/rules"
+	"github.com/terraform-linters/tflint-ruleset-terraform/terraform"
 )
 
 func main() {
 	plugin.Serve(&plugin.ServeOpts{
-		RuleSet: &tflint.BuiltinRuleSet{
-			Name:    "terraform",
-			Version: "0.1.0",
-			Rules: []tflint.Rule{
-				rules.NewTerraformCommentSyntaxRule(),
-				rules.NewTerraformDeprecatedIndexRule(),
-				rules.NewTerraformDeprecatedInterpolationRule(),
-				rules.NewTerraformDocumentedOutputsRule(),
-				rules.NewTerraformDocumentedVariablesRule(),
-				rules.NewTerraformEmptyListEqualityRule(),
-				rules.NewTerraformModulePinnedSourceRule(),
-				rules.NewTerraformModuleVersionRule(),
-				rules.NewTerraformNamingConventionRule(),
-				rules.NewTerraformRequiredProvidersRule(),
-				rules.NewTerraformRequiredVersionRule(),
-				rules.NewTerraformStandardModuleStructureRule(),
-				rules.NewTerraformTypedVariablesRule(),
-				rules.NewTerraformUnusedDeclarationsRule(),
-				rules.NewTerraformUnusedRequiredProvidersRule(),
-				rules.NewTerraformWorkspaceRemoteRule(),
+		RuleSet: &terraform.RuleSet{
+			BuiltinRuleSet: tflint.BuiltinRuleSet{
+				Name:    "terraform",
+				Version: "0.1.0",
+			},
+			PresetRules: map[string][]tflint.Rule{
+				"all": {
+					rules.NewTerraformCommentSyntaxRule(),
+					rules.NewTerraformDeprecatedIndexRule(),
+					rules.NewTerraformDeprecatedInterpolationRule(),
+					rules.NewTerraformDocumentedOutputsRule(),
+					rules.NewTerraformDocumentedVariablesRule(),
+					rules.NewTerraformEmptyListEqualityRule(),
+					rules.NewTerraformModulePinnedSourceRule(),
+					rules.NewTerraformModuleVersionRule(),
+					rules.NewTerraformNamingConventionRule(),
+					rules.NewTerraformRequiredProvidersRule(),
+					rules.NewTerraformRequiredVersionRule(),
+					rules.NewTerraformStandardModuleStructureRule(),
+					rules.NewTerraformTypedVariablesRule(),
+					rules.NewTerraformUnusedDeclarationsRule(),
+					rules.NewTerraformUnusedRequiredProvidersRule(),
+					rules.NewTerraformWorkspaceRemoteRule(),
+				},
+				"recommended": {
+					rules.NewTerraformDeprecatedIndexRule(),
+					rules.NewTerraformDeprecatedInterpolationRule(),
+					rules.NewTerraformEmptyListEqualityRule(),
+					rules.NewTerraformModulePinnedSourceRule(),
+					rules.NewTerraformModuleVersionRule(),
+					rules.NewTerraformRequiredProvidersRule(),
+					rules.NewTerraformRequiredVersionRule(),
+					rules.NewTerraformTypedVariablesRule(),
+					rules.NewTerraformUnusedDeclarationsRule(),
+					rules.NewTerraformUnusedRequiredProvidersRule(),
+					rules.NewTerraformWorkspaceRemoteRule(),
```

Comment:
Unlike the current settings, the following are enabled by default:

- `terraform_deprecated_index`
- `terraform_required_providers`
- `terraform_required_version`
- `terraform_typed_variables`
- `terraform_unused_declarations`
- `terraform_unused_required_providers`

These are the rules that I consider to be recommended even for senior engineers by my standards.

However, there are issues with https://github.com/terraform-linters/tflint/issues/1328 and https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_unused_declarations.md#how-to-fix, and the checks aren't perfect, so I'm not sure if these should be the default.

@bendrucker What do you think? If you have any other thoughts on the default rules, please let me know.

Reply:
Yeah I'd say `terraform_unused_required_providers` needs to have the issue of implicit provider passing into modules addressed before it comes on by default. The others seem like good defaults for a "recommended" config.

Reply:
OK, we will not include `terraform_unused_required_providers` in the default for now.